packages/nimble-components/src/table/index.ts

```diff
@@ -731,6 +760,14 @@ export class Table<
         this.tableUpdateTracker.trackColumnInstancesChanged();
     }
 
+    private removeActionMenuSlotsFromColumnRequestedSlots(): void {
+        for (const actionMenuSlot of this.actionMenuSlots) {
+            this.columnRequestedSlots.delete(actionMenuSlot);
+        }
+
+        this.refreshRows();
+    }
```

Comment:
I'm having a hard time understanding why the columnRequestedSlots may contain actionMenuSlots.
Is this a naming conflict guard? If so can that be checked at validation instead?

Reply:
Yes, it's to allow the same menu/slot to be used for both a menu-button column and an action menu. We could decide that we didn't want to support that and make the table invalid in that case, but it isn't clear to me why that would be better than allowing a single menu to be used for both. The other alternative to having this specific code is that I could look into rewriting some of the action menu slotting code to reuse some of the new slot tracking code that is being added in this PR.

Reply:
> but it isn't clear to me why that would be better than allowing a single menu to be used for both.

yea, fair

> The other alternative to having this specific code is that I could look into rewriting some of the action menu slotting code to reuse some of the new slot tracking code that is being added in this PR.

that does sound nice, I was on the fence about tableData.slots not being used for both or using tableData to hold both (tableData.columnSlots + tableData.actionMenuSlots) maybe as a tech debt and follow-on instead of holding this PR?

Reply:
Created https://github.com/ni/nimble/issues/2248 to track this.